contracts/script/ZKUpgrader.s.sol

```diff
@@ -0,0 +1,163 @@
+
+    ////////////////////////////////////////////////////////////////
+    //                        Functions                           //
+    ////////////////////////////////////////////////////////////////
+
+    function run() public broadcast {
+        upgradeToZK(address(0));
+    }
+
+    function upgradeToZK(address spoofedAdmin) public {
+
+        /////////////////////////////
+        //          INPUTS         //
+        /////////////////////////////
+
+        // grab the config from the zkconfig.json file
+        Config memory config = readJson("zkconfig.json");
+
+        /////////////////////////////
+        //      SAFETY CHECKS      //
+        /////////////////////////////
+
+        // requier that the verifier gateway is deployed
+        require(address(config.verifierGateway).code.length > 0, "ZKUpgrader: verifier gateway not deployed");
+
+        /////////////////////////////
+        // OUTPUT ROOT COMPUTATION //
+        /////////////////////////////
+
+        (bytes32 startingOutputRoot, uint startingTimestamp) = fetchOutputRoot(config);
+
+        /////////////////////////////
+        //     CONTRACT UPGRADE    //
+        /////////////////////////////
+
+        // deploy an implementation of the ZK L2OO to the L1
+        ZKL2OutputOracle zkL2OutputOracleImpl = new ZKL2OutputOracle();
+
+        ZKL2OutputOracle.ZKInitParams memory zkInitParams = ZKL2OutputOracle.ZKInitParams({
+            chainId: config.chainId,
+            verifierGateway: config.verifierGateway,
+            vkey: config.vkey,
+            owner: config.owner,
+            startingOutputRoot: startingOutputRoot
+        });
+
+        if (spoofedAdmin != address(0)) vm.startPrank(spoofedAdmin);
+
+        // upgrade the proxy to the new implementation
+        Proxy(payable(config.l2OutputOracleProxy)).upgradeToAndCall(
+            address(zkL2OutputOracleImpl),
+            abi.encodeCall(ZKL2OutputOracle.initialize, (
+                config.submissionInterval,
+                config.l2BlockTime,
+                config.startingBlockNumber,
+                startingTimestamp,
+                config.proposer,
+                config.challenger,
+                config.finalizationPeriod,
+                zkInitParams
+            ))
+        );
+    }
+
+
+    ////////////////////////////////////////////////////////////////
+    //                         Helpers                            //
+    ////////////////////////////////////////////////////////////////
+
+    function readJson(string memory filepath) view public returns (Config memory) {
+        string memory root = vm.projectRoot();
+        string memory path = string.concat(root, "/", filepath);
+        string memory json = vm.readFile(path);
+        bytes memory data = vm.parseJson(json);
+        return abi.decode(data, (Config));
+    }
+
+    function fetchOutputRoot(Config memory config) public returns (bytes32 startingOutputRoot, uint startingTimestamp) {
```

Comment:
Can you make a new function that calls `setOutputRoot` on a `ZKL2OO` which takes in a block number? Would be similar to this function and is useful for development.